tests/auto/searchmodel/tst_searchmodel.cpp

```diff
@@ -524,5 +527,51 @@ void TestSearchModel::testCaseInsensitiveFilter() {
   QCOMPARE(typeH, "Hero");
 }
 
+void TestSearchModel::testIndexOf() {
```

Comment:
Would be great if could you add some comments explaining exactly what we're testing here.